ocaml/typing/typeclass.ml

```diff
@@ -1327,21 +1337,64 @@ and class_expr_aux cl_num val_env met_env virt self_scope scl =
                    Jkind.Sort.value
                   )
             in
+            let eliminate_position_arg () =
+              (* XXX jrodri: I _think_ this is correct, but I wanted to sanity check.
+                 There are two environments, [val_env] and [met_env], I _think_ it is
+                 correct to pass in [val_env] as IIUC, it is the environment from
+                 "outside" the class (or at least this is what I think as Pcl_constr looks
+                 at the [val_env] instead of the [met_env])?
+ 
+                 jrodri: I am not 100% confident though and it's more of an assumption
+                 that I wanted to sanity check. Is [val_env] correct here? I also wanted
+                 to write a test, but since I _think_ that Texp_src_pos does not refer to
+                 anything from the environment in translcore.ml nor in other parts, I
+                 couldn't think of a way of meaninfully testing/knowing whether I sent in
+                 the wrong environment or not...
+
+                 jrodri: The test I wanted to write was something like attempting to refer to "self" 
+                 from somewhere invalid/valid (as seen in the example below), but I don't
+                 think the src_pos that is written refers to anything...:
+
+                 Invalid:
+                 {v
+─( 17:08:24 )─< command 5 >────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
+utop # let o = object(self)
+  inherit c (if self = self then 1 else 0)
+end;;
+Line 2, characters 16-20:
+Error: The self variable self
+       cannot be accessed from the definition of an instance variable
+                 v}
+
+                 Valid:
+
+                 {v
+─( 17:08:24 )─< command 5 >────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────{ counter: 0 }─
+utop # let o = object(self)
+  method foo = if self = self then 1 else 0
+end;;
+val o : < foo : int; x : int > = <obj>
+                 v}
+              *)
+              let arg = Typecore.src_pos (Location.ghostify scl.pcl_loc) [] val_env in
+              Arg (arg, Jkind.Sort.value)
+            in
             let remaining_sargs, arg =
               if ignore_labels then begin
                 match sargs with
                 | [] -> assert false
                 | (l', sarg) :: remaining_sargs ->
+                    let label_is_absent_in_remaining_args =
+                      lazy (not (List.exists (fun (l, _) -> name = Btype.label_name l) remaining_sargs))
```

Comment:
Instead of using `lazy`, I think it's conceptually simpler just to make `label_is_absent_in_remaining_args` a function that takes `unit`. The function is only ever called at most once.

Reply:
Changed! + also waited for CI to pass again which it did!